src/chart/animator/animation.cpp

```diff
@@ -110,23 +110,54 @@ void Animation::addKeyframe(const Gen::PlotPtr &next,
 	}
 
 	auto begin = target;
+
+	std::size_t real_animation{};
+
 	if (intermediate0) {
-		addKeyframe(target,
+		real_animation += strategy == RegroupStrategy::fade
```

Comment:
This is duplicated, we calculate this here and then in addKeyframe. I think we should remove it from addKeyframe and pass this as an argument instead.